Tests/HttpUnitTests/HttpClientTest.cs

```diff
@@ -0,0 +1,142 @@
+        {
+            HttpClient client = CreateHttpClient();
+            client.Dispose();
+            client.Dispose();
+        }
+
+        [TestMethod]
+        public void DefaultRequestHeaders_Idempotent()
+        {
+            using (HttpClient client = CreateHttpClient())
+            {
+                Assert.NotNull(client.DefaultRequestHeaders);
+                Assert.Same(client.DefaultRequestHeaders, client.DefaultRequestHeaders);
+            }
+        }
+
+        [TestMethod]
+        public void BaseAddress_Roundtrip_Equal()
+        {
+            using (HttpClient client = CreateHttpClient())
+            {
+                Assert.Null(client.BaseAddress);
+
+                Uri uri = new Uri(CreateFakeUri());
+                client.BaseAddress = uri;
+                Assert.Equal(uri.ToString(), client.BaseAddress.ToString());
+
+                client.BaseAddress = null;
+                Assert.Null(client.BaseAddress);
+            }
+        }
+
+        [TestMethod]
+        public void BaseAddress_InvalidUri_Throws()
+        {
+            using (HttpClient client = CreateHttpClient())
+            {
+                Assert.Throws(typeof(ArgumentException),
+                    () => client.BaseAddress = new Uri("ftp://onlyhttpsupported"));
+
+                Assert.Throws(typeof(ArgumentException),
+                    () => client.BaseAddress = new Uri("/onlyabsolutesupported", UriKind.Relative));
+            }
+        }
+
+        [TestMethod]
+        public void Timeout_Roundtrip_Equal()
+        {
+            using (HttpClient client = CreateHttpClient())
+            {
+                client.Timeout = Timeout.InfiniteTimeSpan;
+                Assert.Equal(Timeout.InfiniteTimeSpan.Ticks, client.Timeout.Ticks);
+
+                client.Timeout = TimeSpan.FromSeconds(1);
+                Assert.Equal(TimeSpan.FromSeconds(1).Ticks, client.Timeout.Ticks);
+            }
+        }
+
+        [TestMethod]
+        public void Timeout_OutOfRange_Throws()
+        {
+            using (HttpClient client = CreateHttpClient())
+            {
+                Assert.Throws(typeof(ArgumentOutOfRangeException),
+                    () => client.Timeout = TimeSpan.FromSeconds(-2));
+                Assert.Throws(typeof(ArgumentOutOfRangeException), 
+                    () => client.Timeout = TimeSpan.FromSeconds(0));
+                Assert.Throws(typeof(ArgumentOutOfRangeException),
+                    () => client.Timeout = TimeSpan.FromSeconds(int.MaxValue));
+            }
+        }
+
+        [TestMethod]
+        public void MaxResponseContentBufferSize_ThrowsIfTooSmallForContent(int maxSize, int contentLength, bool exceptionExpected)
+        {
+            maxSize = 1;
+            contentLength = 2;
+            exceptionExpected = true;
+
+            using HttpClient client = CreateHttpClient();
+
+            //LoopbackServer server = new LoopbackServer();
```

Comment:
even in the tests, better to keep the comments clean

Reply:
Sure! This one was left behind because at some point I really had to remove all the test that where depending on network.